### DISCUSSION TITLE
Loading `Material` asset

### DISCUSSION THREAD

**Discussion by Omararafa** [nxrighthere/UnrealCLR#261] [Help] [answered]
I am not able to load my material instance as the only option I have is to load the material, is there an option to load an existing material instance?
![2022-06-27 21_43_31-DayTime (Running) - Microsoft Visual Studio](https://user-images.githubusercontent.com/48950220/176022859-91b1d193-feda-440f-a9a2-228de4846e3a.png)

**Comment by nxrighthere** [comment, marked as answer]
Since you want to work with a material asset, you have to use [`Material`](https://github.com/nxrighthere/UnrealCLR/blob/master/API/Material.md) class for such purposes. Please, see [this](https://github.com/nxrighthere/UnrealCLR/blob/master/Source/Managed/Tests/TextureAssets.cs) test.

[`MaterialInstance`](https://github.com/nxrighthere/UnrealCLR/blob/master/API/MaterialInstance.md) is an abstract class primarily used in [`MaterialInstanceDynamic`](https://github.com/nxrighthere/UnrealCLR/blob/master/API/MaterialInstanceDynamic.md) for the creation of dynamic materials where you can set a scalar, vector, or texture parameter that can be loaded from disk or created from an image asset or a buffer. Please, see [this](https://github.com/nxrighthere/UnrealCLR/blob/6168bb6acd3a455cca209aefac8f5b2cfbb801d1/Source/Managed/Tests/SkeletalMeshes.cs#L34) test.
